lib/cheetah.rb

```diff
@@ -375,6 +378,7 @@ class << self
 
     def run(*args)
       options = args.last.is_a?(Hash) ? args.pop : {}
+      return changed_env(options, *args) if options[:env]
```

Comment:
At this point, `options` is still being set up. Regardless of the way in which we end up processing `:env`, it should be done **after** the defaults are merged in the next statement,